### ISSUE TITLE
LibJS: property_descriptor.has_value() assertion failure in Object::set_integrity_level

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#6643]
Found with Fuzzilli.

Code:
```js
function main() {
const v3 = [1337,1337,1337];
const v5 = Object(..."hasInstance",...v3);
const v6 = Object.freeze(v5);
}
main();
// STDERR:
// FuzzilliJs: ../Userland/Libraries/LibJS/Runtime/Object.cpp:196: bool JS::Object::set_integrity_level(JS::Object::IntegrityLevel): Assertion `property_descriptor.has_value()' failed.
```

Trace:
```
js(26): USERSPACE(26) ASSERTION FAILED: property_descriptor.has_value()
../../Userland/Libraries/LibJS/Runtime/Object.cpp:196
CrashDaemon(15): --- Backtrace for thread #0 (TID 26) ---
CrashDaemon(15): 0xb6f4e1ce: [libsystem.so] syscall2 +0xe (syscall.cpp:25)
CrashDaemon(15): 0x461ae8b0: [libc.so] kill +0x20 (signal.cpp:21)
CrashDaemon(15): 0x461ae953: [libc.so] raise +0x23 (signal.cpp:34)
CrashDaemon(15): 0x4619c25d: [libc.so] abort +0x2a (stdlib.cpp:232)
CrashDaemon(15): 0x4619cd4b: [libc.so] __assertion_failed +0xeb (Format.h:236)
CrashDaemon(15): 0xa1d6382c: [libjs.so] JS::Object::set_integrity_level(JS::Object::IntegrityLevel) +0xa6c (Atomic.h:206)
CrashDaemon(15): 0xa1d5b56b: [libjs.so] JS::ObjectConstructor::freeze(JS::VM&, JS::GlobalObject&) +0x7b (ObjectConstructor.cpp:171)
CrashDaemon(15): 0xa1d0d00f: [libjs.so] AK::Function<JS::Value (JS::VM&, JS::GlobalObject&)>::CallableWrapper<JS::Value (*)(JS::VM&, JS::GlobalObject&)>::call(JS::VM&, JS::GlobalObject&) const +0x2f (Function.h:111)
CrashDaemon(15): 0xa1d55bbb: [libjs.so] JS::NativeFunction::call() +0x4b (NativeFunction.cpp:44)
CrashDaemon(15): 0xa1db296b: [libjs.so] JS::VM::call_internal(JS::Function&, JS::Value, AK::Optional<JS::MarkedValueList>) +0x78b (Vector.h:156)
CrashDaemon(15): 0xa1cc1f2d: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0xe2d (VM.h:274)
CrashDaemon(15): 0xa1cbcdef: [libjs.so] JS::VariableDeclaration::execute(JS::Interpreter&, JS::GlobalObject&) const +0xff (Atomic.h:246)
CrashDaemon(15): 0xa1cd76d5: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xa5 (Interpreter.cpp:149)
CrashDaemon(15): 0xa1d89b0f: [libjs.so] JS::ScriptFunction::execute_function_body() +0x2ef (ScriptFunction.cpp:130)
CrashDaemon(15): 0xa1d89e8d: [libjs.so] JS::ScriptFunction::call() +0x16d (ScriptFunction.cpp:139)
CrashDaemon(15): 0xa1db296b: [libjs.so] JS::VM::call_internal(JS::Function&, JS::Value, AK::Optional<JS::MarkedValueList>) +0x78b (Vector.h:156)
CrashDaemon(15): 0xa1cc1f2d: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0xe2d (VM.h:274)
CrashDaemon(15): 0xa1cbac9f: [libjs.so] JS::ExpressionStatement::execute(JS::Interpreter&, JS::GlobalObject&) const +0xaf (AST.cpp:50)
CrashDaemon(15): 0xa1cd76d5: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xa5 (Interpreter.cpp:149)
CrashDaemon(15): 0xa1cba8bc: [libjs.so] JS::Program::execute(JS::Interpreter&, JS::GlobalObject&) const +0xac (AST.cpp:50)
CrashDaemon(15): 0xa1cd649b: [libjs.so] JS::Interpreter::run(JS::GlobalObject&, JS::Program const&) +0x28b (VM.h:154)
CrashDaemon(15): 0x2d8a3cbd: [/bin/js] parse_and_run(JS::Interpreter&, AK::StringView const&) +0x11d (js.cpp:484)
CrashDaemon(15): 0x2d89c18b: [/bin/js] main +0xb8b (js.cpp:985)
CrashDaemon(15): 0x2d89c9d8: [/bin/js] _start +0x58 (crt0.cpp:39)
```

**Comment by linusg** [comment]
Wrong issue number in commit :upside_down_face:

**Comment by linusg** [comment]
Fixed in #7962. `getOwnPropertyName()` / `getOwnPropertyDescriptor()` still returns bogus values for a string object, but that's a separate issue.
